src/net_processing.cpp

```diff
@@ -2599,7 +2482,7 @@ void ProcessMessage(
                     pfrom.fDisconnect = true;
                     return;
                 } else if (!fAlreadyHave && !chainman.ActiveChainstate().IsInitialBlockDownload()) {
-                    RequestTx(State(pfrom.GetId()), inv.hash, current_time);
+                    RequestTx(pfrom, inv.hash, current_time);
```

Comment:
4f4f2f2

If `fAlreadyHave==true` call `AlreadyHaveTx` to clean transaction entries early ?

Reply:
I don't think that's necessary - we just have to call `ForgetTx` whenever a transaction *becomes* alreadyhave, and prevent adding announcements for alreadyhave ones. If that's done correctly, an addition `ForgetTx` here would always be a no-op.